source/extensions/tracers/zipkin/span_buffer.cc

```diff
@@ -71,20 +72,44 @@ JsonV2Serializer::JsonV2Serializer(const bool shared_span_context)
     : shared_span_context_{shared_span_context} {}
 
 std::string JsonV2Serializer::serialize(const std::vector<Span>& zipkin_spans) {
-  const std::string serialized_elements =
-      absl::StrJoin(zipkin_spans, ",", [this](std::string* out, const Span& zipkin_span) {
+  Util::Replacements replacements;
+  const std::string serialized_elements = absl::StrJoin(
+      zipkin_spans, ",", [this, &replacements](std::string* out, const Span& zipkin_span) {
+        const auto& replacement_values = replacements;
         absl::StrAppend(
-            out, absl::StrJoin(toListOfSpans(zipkin_span), ",",
-                               [](std::string* element, const ProtobufWkt::Struct& span) {
-                                 absl::StrAppend(element, MessageUtil::getJsonStringFromMessage(
-                                                              span, false, true));
-                               }));
+            out, absl::StrJoin(
+                     toListOfSpans(zipkin_span, replacements), ",",
+                     [&replacement_values](std::string* element, const ProtobufWkt::Struct& span) {
+                       const std::string json = MessageUtil::getJsonStringFromMessage(
+                           span, /* pretty_print */ false,
+                           /* always_print_primitive_fields */ true);
+
+                       // The Zipkin API V2 specification mandates to store timestamp value as int64
+                       // https://github.com/openzipkin/zipkin-api/blob/228fabe660f1b5d1e28eac9df41f7d1deed4a1c2/zipkin2-api.yaml#L447-L463
+                       // (often translated as uint64 in some of the official implementations:
+                       // https://github.com/openzipkin/zipkin-go/blob/62dc8b26c05e0e8b88eb7536eff92498e65bbfc3/model/span.go#L114,
+                       // and see the discussion here:
+                       // https://github.com/openzipkin/zipkin-go/pull/161#issuecomment-598558072),
+                       // however when the timestamp is stored as number value (double) there is a
```

Comment:
```suggestion
                       // https://github.com/openzipkin/zipkin-go/pull/161#issuecomment-598558072).
                       // However, when the timestamp is stored as number value in a protobuf struct, it is stored as a double. Because of how protobuf serializes doubles, there is a
```